base/strings/io.jl

```diff
@@ -299,7 +299,8 @@ function unescape_string(io, s::AbstractString)
                     i = j
                 end
                 if k == 1
-                    throw(ArgumentError("\\x used with no following hex digits in $(repr(s))"))
+                    throw(ArgumentError("invalid $(m == 2 ? "hex (\\x)" :
+                                            "unicode (\\u)") escape sequence used in $(repr(s))"))
```

Comment:
If we're going to be this specific, why not go all the way and differentiate `\u` and `\U` correctly as well?